app/src/main/java/app/grapheneos/pdfviewer/fragment/PasswordPromptFragment.kt

```diff
@@ -0,0 +1,74 @@
+package app.grapheneos.pdfviewer.fragment
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextUtils
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.WindowManager
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import app.grapheneos.pdfviewer.PdfViewer
+import app.grapheneos.pdfviewer.R
+import app.grapheneos.pdfviewer.databinding.PasswordDialogFragmentBinding
+import com.google.android.material.textfield.TextInputEditText
+
+class PasswordPromptFragment : DialogFragment() {
+
+    private lateinit var passwordDialogFragmentBinding: PasswordDialogFragmentBinding
+    private lateinit var passwordEditText : TextInputEditText
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val passwordPrompt = AlertDialog.Builder(requireContext())
+        passwordDialogFragmentBinding =
+            PasswordDialogFragmentBinding.inflate(LayoutInflater.from(requireContext()))
+        passwordEditText = passwordDialogFragmentBinding.pdfPasswordEdittext
+        passwordPrompt.setView(passwordDialogFragmentBinding.root)
+        passwordEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
+            override fun onTextChanged(input: CharSequence, i: Int, i1: Int, i2: Int) {
+                val positiveButton: Button =
+                    (dialog as AlertDialog).getButton(DialogInterface.BUTTON_POSITIVE)
+                positiveButton.isEnabled = input.isNotEmpty()
+            }
+
+            override fun afterTextChanged(editable: Editable) {}
+        })
+        passwordEditText.setOnEditorActionListener { _, _, _ ->
+            if (!TextUtils.isEmpty(passwordEditText.text.toString())) {
+                (activity as PdfViewer).loadPdfWithPassword(passwordEditText.text.toString())
+                dialog?.dismiss()
+            }
+            true
+        }
+        passwordPrompt.setPositiveButton(
+            R.string.open
+        ) { _, _ ->
+            val password = passwordEditText.text.toString()
+            (activity as PdfViewer).loadPdfWithPassword(password)
+        }
+        passwordPrompt.setNegativeButton(
```

Comment:
Don't split function call into multiple lines unless it's long.
Replace with `passwordPrompt.setNegativeButton(R.string.cancel, null)`